tests/unit/test_optimisation_problems/test_base_optimisation_problem.py

```diff
@@ -8,27 +8,37 @@
 
 class TestBaseOptimisationProblem(unittest.TestCase):
     def test_base_optimisation_problem_init(self):
-        optimisation_problem = pbparam.BaseOptimisationProblem()
+        optimisation_problem = pbparam.BaseOptimisationProblem(
+            cost_function=pbparam.MLE()
```

Comment:
Is this to prevent it from crashing if cost function is `None`? I think RMSE is a very safe bet which we could make the default, but not sure if this is good practice or not.

Reply:
Yes the way the code has to be restructured requires this to be set even if not used. If we dont want a cost function we could create a blank function to fill this during testing.

Reply:
Would it be better to use `BaseCostFunction`? It has no functionality implemented, but if it is not called anyway it should work.